addons/hw_drivers/iot_handlers/drivers/KeyboardUSBDriver_L.py

```diff
@@ -299,7 +297,22 @@ def _set_device_type(self, device_type='keyboard'):
                 device.grab()
             self.read_barcode_lock = Lock()
         else:
-            self.device_type = 'keyboard'
+            # Check if the keyboard is a touchscreen
+            udev_output = subprocess.run(
+                ["udevadm", "info", "--export-db"], capture_output=True, text=True, check=True
+            ).stdout.split("\n\n")
+
+            # If the device is a touchscreen, we set the device type to 'touchscreen', and rename the device
+            self.device_type, self.device_name = next(
+                (('touchscreen', line.split('"')[1]) for block in udev_output if "ID_INPUT_TOUCHSCREEN=1" in block
+                for line in block.splitlines() if line.startswith("E: NAME=")),
+                ('keyboard', self.device_name)
+            )
+            if self.device_type == 'touchscreen':
+                display = next((device for device in iot_devices.values() if device.device_type == 'display'), None)
```

Comment:
this doesn't seem to work:
- `set_device_type` is called only during the instantiation. This means that if the display is detected and added to the `iot_devices` after the touchscreen, its subtype will never be set as "touchscreen", which leads to the iot box's `can_be_kiosk` value be set to false in the db.